core/src/mindustry/logic/LExecutor.java

```diff
@@ -1459,15 +1459,19 @@ public void run(LExecutor exec){
                 case ban -> {
                     Object cont = exec.obj(value);
                     if(cont instanceof Block b){
+                        boolean update = !state.rules.bannedBlocks.contains(b);
                         state.rules.bannedBlocks.add(b);
```

Comment:
ObjectSet add and remove returns `true` if the collection of modified, so you can avoid the extra `contains call` and the boolean (`if(bannedBlocks.add(b) && !headless){ /*stuff*/ }`).

Reply:
Done, thanks for this tip